### PR TITLE
Xsiam layout unified

### PR DESCRIPTION
**Pull request by eyalpalo**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Must have
- [ ] Tests
- [ ] Documentation
- [ ] [VS Code Extension](https://docs.gitlab.com/ee/ci/yaml/#tags)
  - [ ] Functionality implemented in the extension - <u>\<link-to-pr-here\></u>
  - [ ] N/A - Functionality cannot be implemented in the extension because <u>\<add-reason-here\></u>

### CODE REVIEW

File: demisto_sdk/commands/common/hook_validations/layout_rule.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        return self._is_valid
+
+    def is_valid_version(self):
+        """
+        May deleted or be edited in the future by the use of XSIAM new content
+        """
+        pass
+
+    @error_codes("LR100")
+    def are_all_fields_exist(self):
```
Comment (dantavori) [inline demisto_sdk/commands/common/hook_validations/layout_rule.py:50]: this better be a schema validation (see ST110)
then we can use it in format as well

**Review by dantavori — CHANGES_REQUESTED** [review]
nice
note that the coverage was decreased significantly, please make sure it's increased

File: demisto_sdk/commands/common/tools.py [hunk 2]
```diff
@@ -2115,6 +2121,8 @@ def _get_file_id(file_type: str, file_content: Dict):
         return file_content.get("id", "")
     elif file_type in ID_IN_COMMONFIELDS:
         return file_content.get("commonfields", {}).get("id")
+    elif file_type == FileType.LAYOUT_RULE:
```
Comment (dantavori) [inline demisto_sdk/commands/common/tools.py:2124]: please update get_display_name() too

File: demisto_sdk/commands/common/schemas/layoutrule.yml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    required: true
+  layout_id:
+    type: str
+    required: true
+  fromVersion:
+    type: str
+    required: true
+  toVersion:
+    type: str
+    required: true
```
Comment (dantavori) [inline demisto_sdk/commands/common/schemas/layoutrule.yml:17]:
```suggestion
  toVersion:
    type: str
```
lmk if I'm unaware of something

File: demisto_sdk/commands/common/hook_validations/layout_rule.py (hunk 1 above)
Comment (dantavori) [reply demisto_sdk/commands/common/hook_validations/layout_rule.py:50]: this validation is redundant when using the schema

File: demisto_sdk/commands/common/schemas/layoutrule.yml [hunk 4]
```diff
@@ -0,0 +1,50 @@
+type: map
```
Comment (dantavori) [inline demisto_sdk/commands/common/schemas/layoutrule.yml:1]: please update constants.SCHEMA_TO_REGEX in order to use the schema, and add a UT in structure_test.py

**Review by dantavori — COMMENTED** [review]
almost there

File: demisto_sdk/commands/common/schemas/layoutrule.yml (hunk 3 above)
Comment (eyalpalo) [reply demisto_sdk/commands/common/schemas/layoutrule.yml]: Removed the toVersion arg

File: CHANGELOG.md [hunk 5]
```diff
@@ -40,6 +40,7 @@
 * Changed the **validate** command to fail on the IN145 error code only when the parameter with type 4 is not hidden.
 * Fixed an issue where downloading content layouts with `detailsV2=None` resulted in an error.
 * Fixed an issue where **xdrctemplate** was missing 'external' prefix.
+* Added support for the **Layout Rule** content type in the id-set and the content graph.
```
Comment (dorschw) [inline CHANGELOG.md:43]: please move it to the top.

**Review by dorschw — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 6]
```diff
@@ -8,6 +8,8 @@
 * Added a UUID to name mapper for **download** it replaces UUIDs with names on all downloaded files.
 * Updated the demisto-py to v3.2.6 which now supports basic proxy authentication.
 * Improved the message shown when using **upload** and overwriting packs.
+* Added support for the **Layout Rule** content type in the id-set and the content graph.
+
```
Comment (dorschw) [inline CHANGELOG.md:12]:
```suggestion
```

File: demisto_sdk/commands/common/hook_validations/layout.py [hunk 7]
```diff
@@ -275,6 +278,59 @@ def is_id_equals_name(self):
...
+            "invTimeline",
+        ]
+        invalid_tabs = ["canvas", "evidenceBoard", "relatedIncidents"]
+        invalid_types_contained = []
+
+        marketplace_versions = get_item_marketplaces(
+            self.file_path, item_data=self.current_file
+        )
+        if MarketplaceVersions.MarketplaceV2.value not in marketplace_versions:
+            return True
```
Comment (dorschw) [inline demisto_sdk/commands/common/hook_validations/layout.py:299]: ??

File: demisto_sdk/commands/common/hook_validations/layout.py [hunk 8]
```diff
@@ -275,6 +278,59 @@ def is_id_equals_name(self):
...
+
+        for key in self.current_file.keys():
+            val = self.current_file.get(key)
```
Comment (dorschw) [inline demisto_sdk/commands/common/hook_validations/layout.py:302]:
```suggestion
        for key, val in self.current_file.items():
```

File: demisto_sdk/commands/common/hook_validations/layout.py [hunk 9]
```diff
@@ -275,6 +278,59 @@ def is_id_equals_name(self):
...
+            if isinstance(val, dict):
+                tabs = val.get("tabs", [])
+                if tabs:
```
Comment (dorschw) [inline demisto_sdk/commands/common/hook_validations/layout.py:305]: no need for this if

File: demisto_sdk/commands/common/hook_validations/layout.py [hunk 10]
```diff
@@ -275,6 +278,59 @@ def is_id_equals_name(self):
...
+                    for tab in tabs:
```
Comment (dorschw) [inline demisto_sdk/commands/common/hook_validations/layout.py:306]:
```suggestion
                for tab in val.get("tabs", []):
```

this will mess up indentations but you get the idea

**Review by dorschw — COMMENTED** [review]
random comments on a small subset

File: demisto_sdk/commands/content_graph/tests/test_data/layoutrule.json [hunk 11]
```diff
@@ -0,0 +1,18 @@
+{
```
Comment (dantavori) [inline demisto_sdk/commands/content_graph/tests/test_data/layoutrule.json:1]: why do we need this file? isn't the default data of TestSuite.LayoutRule enough?

File: demisto_sdk/commands/format/update_generic_json.py [hunk 12]
```diff
@@ -93,6 +93,7 @@ def update_json(
         self.set_fromVersion(
             default_from_version=default_from_version, file_type=file_type
         )
+        self.remove_unnecessary_keys()
```
Comment (dantavori) [inline demisto_sdk/commands/format/update_generic_json.py:96]: there's one in line 92 already, is this intentional?

File: demisto_sdk/commands/validate/tests/validators_test.py [hunk 13]
```diff
@@ -168,7 +169,7 @@ def teardown_class(cls):
         os.remove(CONF_PATH)
         for dir_to_delete in cls.CREATED_DIRS:
             if os.path.exists(dir_to_delete):
-                os.rmdir(dir_to_delete)
+                shutil.rmtree(dir_to_delete)
```
Comment (dantavori) [inline demisto_sdk/commands/validate/tests/validators_test.py:172]: I think this change is redundant (probably others), please go over the PR and revert unnecessary changes

File: demisto_sdk/tests/test_files/DummyXSIAMPack/CorrelationRules/invalid_correlation_rules_has_fromversion.yml [hunk 14]
```diff
@@ -0,0 +1,43 @@
+  
+  alert_category: EXECUTION
```
Comment (dantavori) [inline demisto_sdk/tests/test_files/DummyXSIAMPack/CorrelationRules/invalid_correlation_rules_has_fromversion.yml:2]: remove DummyXSIAMPack dir

File: demisto_sdk/commands/common/hook_validations/layout_rule.py [hunk 15]
```diff
@@ -0,0 +1,43 @@
...
+            print_as_warnings=print_as_warnings,
+            json_file_path=json_file_path,
+        )
+        self._is_valid = True
+
+    def is_valid_file(self, validate_rn=True, is_new_file=False, use_git=False):
+        """
+        Check whether the triggers file is valid or not
+        Note: For now we return True regardless of the item content. More info:
+        https://github.com/demisto/etc/issues/48151#issuecomment-1109660727
```
Comment (dantavori) [inline demisto_sdk/commands/common/hook_validations/layout_rule.py:34]:
```suggestion
        Note: For now we return True regardless of the item content.
```

**Review by dantavori — COMMENTED** [review]
you have some changes "inherited" from an old version of my branch, please revert them.
other than that looks great 👍🏽 ping me when ready for approval

File: demisto_sdk/commands/common/hook_validations/layout.py (hunk 7 above)
Comment (eyalpalo) [reply demisto_sdk/commands/common/hook_validations/layout.py:299]: this validation does not relevant for layouts not for XSIAM

File: demisto_sdk/commands/content_graph/tests/test_data/layoutrule.json (hunk 11 above)
Comment (eyalpalo) [reply demisto_sdk/commands/content_graph/tests/test_data/layoutrule.json:1]: it is used in the ```test_layout_rule_parser``` function to create rule using the TestSuit for convenience.
Remove it and insert the json in the code itself?

**Review by dantavori — APPROVED** [review]
